internal/pkg/utils/testhelper/runner/test.go

```diff
@@ -0,0 +1,602 @@
+	stdout := cmdInOut.StdoutString()
+	stderr := cmdInOut.StderrString()
+
+	expectedCode := cast.ToInt(t.ReadFileFromTestDir("expected-code"))
+	assert.Equal(
+		t.t,
+		expectedCode,
+		exitCode,
+		"Unexpected exit code.\nSTDOUT:\n%s\n\nSTDERR:\n%s\n\n",
+		stdout,
+		stderr,
+	)
+
+	expectedStdout := t.ReadFileFromTestDir("expected-stdout")
+	wildcards.Assert(t.t, expectedStdout, stdout, "Unexpected STDOUT.")
+
+	expectedStderr := t.ReadFileFromTestDir("expected-stderr")
+	wildcards.Assert(t.t, expectedStderr, stderr, "Unexpected STDERR.")
+}
+
+type runAPIServerConfig struct {
+	path                string
+	args                []string
+	envs                map[string]string
+	updateRequestPathFn func(string) string
+}
+
+func (t *Test) runAPIServer(
+	path string,
+	addArgs []string,
+	addEnvs map[string]string,
+	updateRequestPathFn func(string) string,
+) {
+	// Get a free port
+	port, err := getFreePort()
+	if err != nil {
+		t.t.Fatalf("Could not receive a free port: %s", err)
+	}
+	apiURL := fmt.Sprintf("http://localhost:%d", port)
+
+	args := append([]string{fmt.Sprintf("--http-port=%d", port)}, addArgs...)
+
+	// Envs
+	envs := env.Empty()
+	envs.Set("PATH", os.Getenv("PATH")) // nolint:forbidigo
+	envs.Set("KBC_STORAGE_API_HOST", t.project.StorageAPIHost())
+	envs.Set("DATADOG_ENABLED", "false")
+	for k, v := range addEnvs {
+		envs.Set(k, v)
+	}
+
+	// Start API server
+	stdout := newCmdOut()
+	stderr := newCmdOut()
+	cmd := exec.Command(path, args...)
+	cmd.Env = envs.ToSlice()
+	cmd.Stdout = io.MultiWriter(stdout, testhelper.VerboseStdout())
+	cmd.Stderr = io.MultiWriter(stderr, testhelper.VerboseStderr())
+	if err := cmd.Start(); err != nil {
+		t.t.Fatalf("Server failed to start: %s", err)
+	}
+
+	cmdWaitCh := make(chan error, 1)
+	go func() {
+		cmdWaitCh <- cmd.Wait()
+		close(cmdWaitCh)
+	}()
+
+	// Kill API server after test
+	t.t.Cleanup(func() {
+		_ = cmd.Process.Kill()
+	})
+
+	// Wait for API server
+	if err = waitForAPI(cmdWaitCh, apiURL); err != nil {
+		t.t.Fatalf(
+			"Unexpected error while waiting for API: %s\n\nServer STDERR:%s\n\nServer STDOUT:%s\n",
+			err,
+			stderr.String(),
+			stdout.String(),
+		)
+	}
+
+	// Run the requests
+	requestsOk := t.runRequests(apiURL, updateRequestPathFn)
+
+	// Shutdown API server
+	_ = cmd.Process.Signal(syscall.SIGTERM)
+	select {
+	case <-time.After(10 * time.Second):
+		t.t.Fatalf("timeout while waiting for server shutdown")
+	case <-cmdWaitCh:
+		// continue
+	}
+
+	// Dump process stdout/stderr
+	assert.NoError(t.t, t.workingDirFS.WriteFile(filesystem.NewRawFile("process-stdout.txt", stdout.String())))
+	assert.NoError(t.t, t.workingDirFS.WriteFile(filesystem.NewRawFile("process-stderr.txt", stderr.String())))
+
+	// Check API server stdout/stderr
+	if requestsOk {
+		if t.testDirFS.IsFile(expectedStdoutPath) || t.testDirFS.IsFile(expectedStderrPath) {
+			// Wait a while the server logs everything for previous requests.
+			time.Sleep(100 * time.Millisecond)
+		}
```

Comment:
Toto by tu uz nemalo byt potrebne, ... ako je vyssie `shutdown`.

Shutdown sa pridaval az neskor, a toto sa asi zabudo zmazat.

Reply:
Ak to bude robit problemy, tak by sa mal `r.logger.DebugWriter()`, ktory je vyssie, ulozit do premennej a po shutdown by sa malo zavolat `wr.Close()`.

Reply:
Ide o to, aby ti testy nahodne nepadali na tom, ze `100ms` bude malo.